### ISSUE TITLE
[Infra] Makefile to magefile migration

### ISSUE THREAD

**Issue by iajrz** [pokt-network/pocket#43]
The makefile is in place to make it easier and faster to iterate on build target experiments. As those are validated, they would be settling into the magefile. Ideally, we'll start using the magefile as the place for further experiments after the team becomes familiarized with it; in the meantime, we know we need to do the migration (represented by the existence of this issue) and plan to address it as part of the Wiring milestone.

**Comment by Olshansk** [comment]
Wanted to add that the TODOs and interim solution in https://github.com/pokt-network/pocket/pull/29 can act as the "origin document" for this issue.

Also, do you think it's worth updating the issue to the structure outlined in the Contribution guide at https://github.com/pokt-network/pocket/pull/36?

**Comment by Olshansk** [comment]
I read through https://bitfieldconsulting.com/golang/scripting earlier today which discussed this library: https://github.com/bitfield/script.

tl;dr You get all of the simplicity of shell scripting with the power of Go, and together they're actually better without making any tradeoffs.

E.g. You can replace `cut -d' ' -f 1 access.log |sort |uniq -c |sort -rn |head` with `File("log").Column(1).Freq().First(10).Stdout()`. 

We can have a longer discussion about it, but I very much believe this can be the solution to a simple migration while getting the best of both worlds.

**Comment by Olshansk** [comment]
Going to close this out for now as we'll revisit it as the project matures.
